regolith/helpers/u_milestonehelper.py

```diff
@@ -110,161 +113,177 @@ def construct_global_ctx(self):
+            all_miles = []
+            id = []
+            target_mil = fragment_retrieval(self.gtx["projecta"], ["milestones"], rc.milestone_uuid)
+            target_del = fragment_retrieval(self.gtx["projecta"], ["_id"], rc.milestone_uuid)
+            target_ko = fragment_retrieval(self.gtx["projecta"], ["_id"], rc.milestone_uuid[2:])
+            if target_mil:
+                for prum in target_mil:
+                    milestones = prum['milestones']
+                    for milestone in milestones:
+                        if milestone.get('uuid') == rc.milestone_uuid:
+                            upd_mil.append(milestone)
+                            all_miles.append(milestones)
+                    if upd_mil:
+                        pid = prum.get('_id')
+                        id.append(pid)
+            if target_del:
+                for prum in target_del:
+                    if prum.get('_id') == rc.milestone_uuid:
+                        deliverable = prum['deliverable']
+                        upd_mil.append(deliverable)
+                    if upd_mil:
+                        pid = prum.get('_id')
+                        id.append(pid)
+            if target_ko:
+                for prum in target_ko:
+                    if prum.get('_id') == rc.milestone_uuid[2:]:
+                        kickoff = prum['kickoff']
+                        upd_mil.append(kickoff)
+                    if upd_mil:
+                        pid = prum.get('_id')
+                        id.append(pid)
+            pdoc = {}
+            filter_id = {'_id': id[0]}
+            for dict in upd_mil:
+                pdoc.update(dict)
+            if rc.finish:
+                rc.status = "finished"
+                pdoc.update({'end_date': now})
+                notes = pdoc.get("notes", [])
+                notes_with_closed_items = [note.replace('()', '(x)', 1) for note in notes]
+                pdoc["notes"] = notes_with_closed_items
+                if target_mil or target_ko:
+                    print("The milestone {} has been marked as finished in prum {}".format(
+                                                pdoc.get('name'), id[0]))
+                if target_del:
+                    name = 'deliverable'
+                    print("The milestone {} has been marked as finished in prum {}".format(
+                                                name, id[0]))
+            if rc.due_date:
+                pdoc.update({'due_date': rc.due_date})
+            if rc.name and not target_del:
+                pdoc.update({'name': rc.name})
+            if rc.objective and not target_del:
+                pdoc.update({'objective': rc.objective})
+            if rc.status:
+                pdoc.update({'status': rc.status})
+            if rc.audience:
+                pdoc.update({'audience': rc.audience})
+            if rc.notes:
+                pdoc.update({'notes': rc.notes})
+
+#        rc.index = rc.index.replace(" ", "")
```

Comment:
delete all this (you can get it back later with a git checkout if you need it, but you won't...)

Reply:
:+1: